### DISCUSSION TITLE
Lost all data

### DISCUSSION THREAD

**Discussion by rowanmmc** [MicroPad/Web#359] [General]
Hi. I am an independent game designer and I use MicroPad for a lot of my work.

Unfortunately it appears that all of my notebooks are completely gone. I believe it was yesterday that I downloaded a new portable install of MicroPad to try the latest version. Upon opening the new version, all of my data was there. However, today all of my notebooks are completely gone, either using my original install or the new portable one.

Let me make it clear that I don't think this is really anything but my fault. It was foolish to not back up data I valued, it was foolish to not set up MicroPad Sync and it was exceptionally foolish to just go ahead and install a second version of the app at the same time and not expect any issues.

I still wanted to write here though. One, I was wondering if there is anything further I can do to try and recover my data. Looking at the IndexedDB folder, it appears that all of my data has been replaced with folders created upon launching MicroPad today (when everything was cleared).

Two, I simply wanted to make this occurrence known. Would this be expected of having a second install of the app? Was it even caused by the second install? I don't know, but hopefully making you aware that this happened can provide something so this doesn't happen to another person as foolish as me, lol.

I'm kind of expecting that my data will not be recoverable. I still very much believe in this software, though, and I will continue to use it for my work (and I am going to set up Sync immediately). I am working on a post of some of my thoughts for what could be refined further on the app based on my time with it, and I hope to post it on this discussion board on a different thread soon.

Thanks, 

Rowan M.

**Comment by NickGeek** [comment]
Hi,
I’m really sorry to hear that you lost your notes. Arguably the most fundamental feature of any note taking app is data integrity.

It still needs further investigation but what I think happens is when there are two different versions of Electron MicroPad running (as in different versions of Electron, not MicroPad itself), if an older version is opened after a newer version, it clears out the app data. I suspect this is Chromium behaviour to avoid issues when downgrading but that’s purely a guess.

So basically if you have Electron v1 and Electron v2:

1.  Open Electron v1 and add a note
2. Open Electron v2 and view that note
3. Open Electron v1 and all the notes are gone

Sadly the most MicroPad can do is request that its data be considered “permanent”, which it has always done. Relying on Chromium not to clear its app data isn’t as stable as I would like, which is why I wanted to use the file system directly instead of IndexedDB. That couldn’t happen at the time I made MicroPad because the FS API had not been stabilised in any browsers yet.

Desktop Chrome now has support for parts of the File System Access API and I have an issue for trying to use it over IndexedDB (https://github.com/MicroPad/MicroPad-Core/issues/328). Safari, Firefox, and Chrome Mobile still don’t support any of the API.

In the meantime I’d have to highly recommend saving important notebooks to disk and/or using MicroSync. A notebook can be saved to disk in two ways:

- Open the options modal for the notebook (right click on its name in the side-bar or click the cog by its name) and then select “Export Notebook”
- Alternatively, Ctrl+S (for Windows/Linux) and Cmd+S (for macOS/iPad OS) will export the notebook, just like in any other app with save functionality

You can save all notebooks by opening the notebooks dropdown and selecting “Export All” or pressing Ctrl+Shift+S (or Cmd+Shift+S). This will open a modal, if you select the “NPX” option it will create a ZIP file of all your notebooks that can be imported again at a later date:
![IMG_0081](https://user-images.githubusercontent.com/3462055/122395326-f7a41080-cfca-11eb-8815-26ba838d32e6.jpeg)

**Reply by rowanmmc** [reply]
Wow, thank you very much for the fast and comprehensive reply. I really appreciate it.

Thanks for the advice on backups. I will definitely be saving to disk and setting up Sync as well.

**Comment by NickGeek** [comment]
Just to follow up, I can confirm this is Chrome behaviour. Chrome does not allow version downgrades and automatically wipes all saved data if it detects one. That’s a really annoying feature to have with no warning but I assume they have their reasons.
